### ISSUE TITLE
Nil pointer in TipSel

### ISSUE THREAD

**Issue by karimodm** [iotaledger/iota-core#451]
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0xe9f673]

goroutine 168 [running]:
github.com/iotaledger/iota-core/pkg/model.(*Block).ProtocolBlock(...)
        /scratch/pkg/model/block.go:89
github.com/iotaledger/iota-core/pkg/protocol/engine/tipselection/v1.DynamicLivenessThreshold.func1({0x2c0dea0, 0xc01397dc80})
        /scratch/pkg/protocol/engine/tipselection/v1/provider.go:45 +0x33
github.com/iotaledger/iota-core/pkg/protocol/engine/tipselection/v1.(*TipSelection).classifyTip(0xc0003dcb00, {0x2c0dea0, 0xc01397dc80})
        /scratch/pkg/protocol/engine/tipselection/v1/tip_selection.go:166 +0xf9
github.com/iotaledger/hive.go/runtime/event.(*Event1[...]).Trigger.func1(0xc0005cf4a0?)
        /go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231019175611-f85bce9102d1/event/events.go:88 +0x1b4
github.com/iotaledger/hive.go/ds/orderedmap.(*OrderedMap[...]).ForEach(0xc0105e7b48, 0xc0105e7b20?)
        /go/pkg/mod/github.com/iotaledger/hive.go/ds@v0.0.0-20231019175611-f85bce9102d1/orderedmap/orderedmap.go:130 +0xc4
github.com/iotaledger/hive.go/runtime/event.(*Event1[...]).Trigger(0xc039f6a750?, {0x2c0dea0?, 0xc01397dc80?})
        /go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231019175611-f85bce9102d1/event/events.go:70 +0x8a
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).setupBlockMetadata(0xc000394e00, 0xc01397dc80)
        /scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:143 +0x146
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).forEachParentByType(0xc000394e00, 0xc039f82ab0?, 0xc0105e7ca8)
        /scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:164 +0x20e
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).setupBlockMetadata(0xc000394e00, 0xc01397db00)
        /scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:135 +0x125
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).forEachParentByType(0xc000394e00, 0x1be9b168e1074?, 0xc0105e7dd0)
        /scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:164 +0x20e
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).setupBlockMetadata(0xc000394e00, 0xc01397d800)
        /scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:135 +0x125
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).AddBlock(0x922651?, 0xc04147d080)
        /scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:70 +0x13c
github.com/iotaledger/hive.go/lo.Void[...].func1()
        /go/pkg/mod/github.com/iotaledger/hive.go/lo@v0.0.0-20231019175611-f85bce9102d1/lo.go:317 +0x17
github.com/iotaledger/hive.go/runtime/event.(*Event1[...]).Trigger.func1.1()
        /go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231019175611-f85bce9102d1/event/events.go:86 +0x1f
github.com/iotaledger/hive.go/runtime/workerpool.(*Task).run(0xc0308c7ec0)
        /go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231019175611-f85bce9102d1/workerpool/task.go:48 +0x72
github.com/iotaledger/hive.go/runtime/workerpool.(*WorkerPool).workerReadLoop(0xc000215ce0)
        /go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231019175611-f85bce9102d1/workerpool/workerpool.go:190 +0x25
github.com/iotaledger/hive.go/runtime/workerpool.(*WorkerPool).worker(0xc000215ce0)
        /go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231019175611-f85bce9102d1/workerpool/workerpool.go:170 +0x50
created by github.com/iotaledger/hive.go/runtime/workerpool.(*WorkerPool).startWorkers in goroutine 1
        /go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231019175611-f85bce9102d1/workerpool/workerpool.go:162 +0x25
```

----------
----------
----------

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0xe9f673]

goroutine 166 [running]:
github.com/iotaledger/iota-core/pkg/model.(*Block).ProtocolBlock(...)
	/scratch/pkg/model/block.go:89
github.com/iotaledger/iota-core/pkg/protocol/engine/tipselection/v1.DynamicLivenessThreshold.func1({0x2c0dea0, 0xc02888cc00})
	/scratch/pkg/protocol/engine/tipselection/v1/provider.go:45 +0x33
github.com/iotaledger/iota-core/pkg/protocol/engine/tipselection/v1.(*TipSelection).classifyTip(0xc000447080, {0x2c0dea0, 0xc02888cc00})
	/scratch/pkg/protocol/engine/tipselection/v1/tip_selection.go:166 +0xf9
github.com/iotaledger/hive.go/runtime/event.(*Event1[...]).Trigger.func1(0xc0005f2420?)
	/go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231019175611-f85bce9102d1/event/events.go:88 +0x1b4
github.com/iotaledger/hive.go/ds/orderedmap.(*OrderedMap[...]).ForEach(0xc047b7b208, 0xc047b7b1e0?)
	/go/pkg/mod/github.com/iotaledger/hive.go/ds@v0.0.0-20231019175611-f85bce9102d1/orderedmap/orderedmap.go:130 +0xc4
github.com/iotaledger/hive.go/runtime/event.(*Event1[...]).Trigger(0xc00b95d140?, {0x2c0dea0?, 0xc02888cc00?})
	/go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231019175611-f85bce9102d1/event/events.go:70 +0x8a
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).setupBlockMetadata(0xc000404e80, 0xc02888cc00)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:143 +0x146
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).forEachParentByType(0xc000404e80, 0xc00c4caa50?, 0xc047b7b368)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:164 +0x20e
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).setupBlockMetadata(0xc000404e80, 0xc02888ca80)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:135 +0x125
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).forEachParentByType(0xc000404e80, 0xc00dfa9a70?, 0xc047b7b490)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:164 +0x20e
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).setupBlockMetadata(0xc000404e80, 0xc02888c900)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:135 +0x125
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).forEachParentByType(0xc000404e80, 0xc00cdee450?, 0xc047b7b5b8)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:164 +0x20e
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).setupBlockMetadata(0xc000404e80, 0xc02888c780)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:135 +0x125
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).forEachParentByType(0xc000404e80, 0xc00bb7a840?, 0xc047b7b6e0)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:164 +0x20e
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).setupBlockMetadata(0xc000404e80, 0xc02888c600)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:135 +0x125
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).forEachParentByType(0xc000404e80, 0xc00e6b05a0?, 0xc047b7b808)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:164 +0x20e
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).setupBlockMetadata(0xc000404e80, 0xc02888c480)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:135 +0x125
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).forEachParentByType(0xc000404e80, 0xc0069ddb00?, 0xc047b7b930)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:164 +0x20e
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).setupBlockMetadata(0xc000404e80, 0xc02888c300)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:135 +0x125
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).forEachParentByType(0xc000404e80, 0xc00cd9e390?, 0xc047b7ba58)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:164 +0x20e
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).setupBlockMetadata(0xc000404e80, 0xc02888c180)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:135 +0x125
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).forEachParentByType(0xc000404e80, 0xc0085fc8d0?, 0xc047b7bb80)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:164 +0x20e
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).setupBlockMetadata(0xc000404e80, 0xc02888c000)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:135 +0x125
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).forEachParentByType(0xc000404e80, 0xc010367770?, 0xc047b7bca8)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:164 +0x20e
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).setupBlockMetadata(0xc000404e80, 0xc0274b9e00)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:135 +0x125
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).forEachParentByType(0xc000404e80, 0x1be9b168e1074?, 0xc047b7bdd0)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:164 +0x20e
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).setupBlockMetadata(0xc000404e80, 0xc0274b9b00)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:135 +0x125
github.com/iotaledger/iota-core/pkg/protocol/engine/tipmanager/v1.(*TipManager).AddBlock(0x922651?, 0xc03fd4af20)
	/scratch/pkg/protocol/engine/tipmanager/v1/tipmanager.go:70 +0x13c
github.com/iotaledger/hive.go/lo.Void[...].func1()
	/go/pkg/mod/github.com/iotaledger/hive.go/lo@v0.0.0-20231019175611-f85bce9102d1/lo.go:317 +0x17
github.com/iotaledger/hive.go/runtime/event.(*Event1[...]).Trigger.func1.1()
	/go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231019175611-f85bce9102d1/event/events.go:86 +0x1f
github.com/iotaledger/hive.go/runtime/workerpool.(*Task).run(0xc00c97c5a0)
	/go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231019175611-f85bce9102d1/workerpool/task.go:48 +0x72
github.com/iotaledger/hive.go/runtime/workerpool.(*WorkerPool).workerReadLoop(0xc000257340)
	/go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231019175611-f85bce9102d1/workerpool/workerpool.go:190 +0x25
github.com/iotaledger/hive.go/runtime/workerpool.(*WorkerPool).worker(0xc000257340)
	/go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231019175611-f85bce9102d1/workerpool/workerpool.go:170 +0x50
created by github.com/iotaledger/hive.go/runtime/workerpool.(*WorkerPool).startWorkers in goroutine 1
	/go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231019175611-f85bce9102d1/workerpool/workerpool.go:162 +0x25
```

**Comment by piotrm50** [comment]
Created a PR that removes some unnecessary checks and prints a block that could cause the nil pointer panic. Need to run the network and investigate further once it happens and we learn more about it from the added log.

https://github.com/iotaledger/iota-core/pull/489

**Comment by jonastheis** [comment]
Should be fixed with 48b1db2eb87d5717fe5cec53a0527046c7babee9
